libraries/AP_Mission/AP_Mission.cpp

```diff
@@ -2025,6 +2063,33 @@ bool AP_Mission::get_next_do_cmd(uint16_t start_index, Mission_Command& cmd)
 /// jump handling methods
 ///
 
+// Set the mission index to the first JUMP_TAG with this tag.
+// Returns true on success, else false if no appropriate JUMP_TAG match can be found or if setting the index failed
+bool AP_Mission::jump_to_tag(const uint16_t tag)
+{
+    const uint16_t index = get_index_of_jump_tag(tag);
+    if (index == 0) {
+        return false;
+    }
+    return set_current_cmd(index);
+}
+
+// find the first JUMP_TAG with this tag and return its index.
+// Returns 0 if no appropriate JUMP_TAG match can be found.
+uint16_t AP_Mission::get_index_of_jump_tag(const uint16_t tag) const
+{
+    for (uint16_t i = 1; i < (unsigned)_cmd_total; i++) {
```

Comment:
this needs the fast search change from sdcard PR

Reply:
Here's the 2-line PR https://github.com/ArduPilot/ardupilot/pull/22996 to do that